### PR TITLE
[DEMO] DependencyController

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Moved all endpoints that are connected with files to `DependencyController` (`/demo/internal/files`)
 * Changed base path for `ManagementController` (`/demo/internal/manager`)
 * Fixed all existing demo request urls

### CODE REVIEW

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        .flatMapIterable { demo ->
+            fileDtos.map { Dependency(demo, version, it.name, it.requiredId()) }
+        }
+        .filterWhen {
+            dependencyStorage.doesExist(it).map(Boolean::not)
+        }
+        .collectList()
+        .map { dependencies ->
+            downloadToolService.downloadToStorage(dependencies).let { size ->
+                StringResponse("Successfully saved $size files to demo storage.", HttpStatus.OK)
```
Comment (nulls) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:99]: maybe add `StringResponse` when all `dependencies` already in storage? (all entities are filtered out by `dependencyStorage.doesExist(it).map(Boolean::not)`)

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt [hunk 2]
```diff
@@ -5,89 +5,74 @@ import com.saveourtool.save.demo.DemoInfo
...
-    @PostMapping("/add-tool")
-    fun addTool(@RequestBody demoDto: DemoDto): Mono<DemoDto> = demoDto.toMono()
+    @PostMapping("/add")
+    fun add(@RequestBody demoDto: DemoDto): Mono<DemoDto> = demoDto.toMono()
         .asyncEffect { downloadToolService.initializeGithubDownload(it.githubProjectCoordinates, it.vcsTagName) }
         .requireOrSwitchToResponseException({ validate() }, HttpStatus.CONFLICT) {
             "Demo creation request is invalid: fill project coordinates, run command and file name."
         }
-        .asyncEffect { blockingToMono { demoService.saveIfNotPresent(it.toDemo()) } }
+        .map { demoService.saveIfNotPresent(it.toDemo()).toDto() }
```
Comment (nulls) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt:37]: better to use `flatMap { blockingToMono { demoService.saveIfNotPresent(it.toDemo()).toDto() } } `. I guess `saveIfNotPresent` is database operations and better to wrap by additional scheduler

Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt:37]: Oh, will do in next PR then

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:99]: Also will add in next PR as this one is already merged

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt (hunk 2 above)
Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/ManagementController.kt:37]: https://github.com/saveourtool/save-cloud/pull/1852

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DependencyController.kt:99]: https://github.com/saveourtool/save-cloud/pull/1852

**Pull request merged**